index.js

```diff
@@ -186,9 +214,21 @@ function buildMemoizer (options) {
                 return cacheAndReturn()
               }
 
+              function hasOwn (obj, key) {
+                return Object.prototype.hasOwnProperty.call(obj, key)
+              }
+
               function parseResult (resultString) {
                 try {
-                  return JSON.parse(resultString).data // will fail on NaN
+                  const deserializedValue = optExt.deserialize(resultString)
+
+                  // NOTE not needed to be tested it's simple feature
+                  // allowing users with custom serialize/deserialize
+                  // to not need to return { data: they result}, but instead directly their result
+                  /* istanbul ignore next */
```

Comment:
We don’t have to have 100% coverage. I’d rather see clearly what is tested and what is not.

Reply:
This might be problematic if we are memoizing a function that actually returns an object containing a `data` property. So I guess we have to come up with a safer approach here.

Reply:
Have you had a look at https://github.com/yahoo/serialize-javascript ?

Reply:
> So I guess we have to come up with a safer approach here.

True. I thought to put `.data` in the default `deserialize` which will mean it is the default behavior, but when use pass deserialize its return is considered entirely.

Reply:
Yea, it looks good. :)